### DISCUSSION TITLE
Need help setting up a RLS policy to allow "semi-anonymous" post/insert on a table

### DISCUSSION THREAD

**Discussion by joakimgk** [supabase/supabase#11690] [Questions] [answered]
I'm struggling with a RLS policy to allow anonymous INSERT (post) over HTTPS.
My toy idea is to have a `project` table, which associates a "passhash" with a project id:

|id int8|passhash varchar|
|-------|-------------------|
|1|abc|
|2|cde|

Then I want to insert into this table, with the following policy (RLS enabled):

|id int8|project_id int8|status varchar|
|-------|------------|-------|

CHECK expression (for INSERT):
```
((getpasshash2())::text = (( SELECT DISTINCT project.passhash
   FROM project
  WHERE (project.id = 1)))
)
```

Here, I've only hard coded the project id (1), but the idea is for this also to be given in a header. The HTTPS header contains the `apikey` and the `device_key` field, with value "abc". I keep getting 401 errors in Postman:

```
 {
    "code": "42501",
    "details": null,
    "hint": null,
    "message": "new row violates row-level security policy for table \"tick\""
}
```

The `getpasshash2()` function looks like this (thanks to @GaryAustin1):

```
  select nullif(current_setting('request.headers', true)::json->>'device_key', '')::varchar;
```

**Comment by joakimgk** [comment]
Forgot to say, the function seems to be working, getting the `device_id` (passhash) text from the HTTPS headers. So this policy will work:

```
((getpasshash2())::text = "abc"::text
)
```
Then I get a HTTP 201 Created. Apparently, there is something with the right-hand side of the policy, the query on the projects table.

**Comment by GaryAustin1** [comment, marked as answer]
Do you have RLS on the projects table?  You will need to meet that select policy when running your RLS above.

**Reply by joakimgk** [reply]
Oh man... That must be it! I would not had thought of that, but I'm sure it's what's missing. Thanks again!
